idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/asJava/firLightUtils.kt

```diff
@@ -120,7 +120,7 @@ internal fun ConeKotlinType.asPsiType(
     mode: TypeMappingMode,
     psiContext: PsiElement,
 ): PsiType {
-
+    if (this.isUnit) return PsiType.VOID
```

Comment:
It looks like this behaviour is incorrect. Consider the following case:
```
fun foo(a: Unit){}
```
for this code, Kotlin compiler will generate the following JVM signature
```
foo(Lkotlin/Unit;)V // Unit is still here
```

Because of that cahnge, some FIR light classes tests fails (see FirLightClassTestGenerated.testAnnotatedPropertyWithSites, FirLightClassTestGenerated.NullabilityAnnotations.testUnitAsGenericArgument, FirLightClassTestGenerated.NullabilityAnnotations.testUnitParameter)

`void` should be used only as a return type of Kotlin "prodcedure" condireing logic of Kotlin light classes: https://github.com/JetBrains/kotlin/blob/111e54c8c1b523472d36f7a316ca9acb90086f9b/compiler/light-classes/src/org/jetbrains/kotlin/asJava/classes/ultraLightMembersCreator.kt#L231-L234

Reply:
Maybe the same behaviour should be sued for the uast too.

Reply:
I saw something similar is done: https://github.com/JetBrains/kotlin/blob/master/idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/asJava/methods/FirLightSimpleMethodForSymbol.kt#L129 but maybe not exactly same as it does not check if the corresponding method is Kotlin "procedure".

Anyway, I don't think I fully understood the details of `PsiType` conversion. I have to admit I naively attempted to remove the gap regarding UAST type results. Let me revisit this later once tree structure conversion is done. I simply dropped the last commit, which wasn't scope of this PR anyway. (Also dropped the test update in FIR UAST side.)